### DISCUSSION TITLE
Kibana "Unknown field"

### DISCUSSION THREAD

**Discussion by tomjones1977** [Security-Onion-Solutions/securityonion#6083] [Unsupported Versions] [answered]
Hello all,

I have enabled some filebeat modules and custom elasticsearch ingest pipelines that utilise fields that don't appear to be part of the index fields.  In the old SecOnion you could go to index management and refresh to start indexing those fields for future data.  However there doesn't seem to be an option to enable this on the new version.  

What is the most appropriate way to ingest and index new fields that aren't standard?

**Comment by weslambert** [comment, marked as answer]
To clarify, you are seeing them in Hunt/Kibana, but the fields themselves aren't being indexed?

In order for a field to be indexed, Elasticsearch must know what data type it should use when indexing the field.  To do this, it references an index template which provides those definitions/mappings.  

Can you provide an example of some fields that aren't being indexed?

All of the default Elasticsearch templates reside in `/opt/so/saltstack/default/salt/elasticsearch/templates/so`, all custom  templates should be placed in `/opt/so/saltstack/local/salt/elasticsearch/templates/custom`, and all modified templates should be placed in `/opt/so/saltstack/local/salt/elasticsearch/templates/so`. 

If you do decide to use a custom template, you'll also need to make sure it's referenced in the `search.sls` so that it is picked up and used by Elasticsearch.  To do this, add the `custom/` prefix and the name of your template file into `search.sls` (after copying from `/opt/so/salstack/default/pillar/elasticsearch/` to `/opt/so/saltstack/local/pillar/elasticsearch`), then restart Elasticsearch with `so-elasticsearch-restart` on the search node(s).

**Reply by sinnetll19** [reply]
can you provide an example of what should be included in the custom template in order to get a specific field index. for example i am using the filebeat panw module and it does not index the field panw.panos.action. when adding the field through the kibana -> stack management -> index pattern -> add field to *:so-* it works temporarily until it gets overwritten by salt i assume.

**Reply by tomjones1977** [reply]
One example is in the screenshot below.  These fields are (from what I can see) created by filebeat so I assume they should be part of the index without manual intervention?

![image](https://user-images.githubusercontent.com/32241793/143439763-de9c122c-7a14-49cc-bfdd-154884abe32c.png)

**Reply by ddbrandonb** [reply]
![image](https://user-images.githubusercontent.com/57487812/146253082-5c3f8ab7-72fc-4e50-976b-00a2b98366d5.png)
I am also seeing this problem, some more of the fields not properly being mapped above.

**Reply by weslambert** [reply]
You'll need to copy and modify the common template or the Checkpoint template to have different mappings be applied to these indices. You can copy the default template from `/opt/so/saltstack/default/salt/elasticsearch/templates/so/so-common-template.json.jinja` (or `so-checkpoint-template.json.jinja`) to `/opt/so/saltstack/local/salt/elasticsearch/templates/so/` and make your changes there.
